watertap/unit_models/electrodialysis_1D.py

```diff
@@ -1556,6 +1690,170 @@ def eq_dl_thickness(self, mem, side, t, x):
+            )
+
+        @self.Constraint(
+            doc="To calculate Sc",
+        )
+        def eq_Sh(self):
+
+            return self.Sh == 0.29 * self.Re**0.5 * self.Sc**0.33
+
+    def _pressure_drop_calculation(self):
+        self.pressure_drop = Var(
+            self.flowsheet().time,
+            initialize=1e4,
+            units=pyunits.pascal * pyunits.meter**-1,
+            doc="pressure drop per unit of length in a single ED channel",
+        )
+        self.pressure_drop_total = Var(
+            self.flowsheet().time,
+            initialize=1e6,
+            units=pyunits.pascal,
+            doc="pressure drop over an entire ED stack",
+        )
+        self.friction_factor = Var(
+            initialize=10,
+            units=pyunits.dimensionless,
+            doc="friction factor of the channel fluid",
+        )
+
+        if self.config.pressure_drop_method == PressureDropMethod.experimental:
+            _log.warning(
+                "Do not forget to FIX the experimental pressure drop value in [Pa/m]!"
+            )
+        else:  # PressureDropMethod.Darcy_Weisbach is used
+            self._get_fluid_dimensionless_quantities()
+
+            @self.Constraint(
+                self.flowsheet().time,
+                doc="To calculate pressure drop per unit length in a single channel",
+            )
+            def eq_pressure_drop(self, t):
+                return (
+                    self.pressure_drop[t]
+                    == self.dens_mass
+                    * self.friction_factor
+                    * self.velocity_D[0, 0] ** 2
+                    * 0.5
+                    * self.hydraulic_diameter**-1
+                )
+
+            if self.config.friction_factor_method == FrictionFactorMethod.defined:
+                _log.warning("Do not forget to FIX the Darcy's friction factor value!")
+            else:
+
+                @self.Constraint(
+                    doc="To calculate friction factor",
+                )
+                def eq_friction_factor(self):
+                    if (
+                        self.config.friction_factor_method
+                        == FrictionFactorMethod.Ponzio
+                    ):
+                        if value(self.Re) < 61:
+                            return self.friction_factor == 1400 * self.Re**-1
+                        else:
+                            return self.friction_factor == 104.5 * self.Re**-0.37
+                    elif (
+                        self.config.friction_factor_method
+                        == FrictionFactorMethod.Gurreri
+                    ):
+                        return (
+                            self.friction_factor
+                            == 4 * 50.6 * self.spacer_porosity**-7.06 * self.Re**-1
+                        )
+                    elif (
+                        self.config.friction_factor_method
+                        == FrictionFactorMethod.Kuroda
+                    ):
```

Comment:
I really like that you added more options for calculating the friction factor, which is something we might want to do in RO later on. Just a comment of admiration =)

Reply:
Thank you!  This might reflect somewhat my hard time in choosing things =)